documentation/modules/appendix/pages/best-practices.adoc

```diff
@@ -12,56 +12,854 @@ SPDX-License-Identifier: MPL-2.0
+
+|===
+|SAMM `name` |Resulting name in generated code (automated case conversion)
+
+|{nok-small} `documentURL`
+|`DOCUMENT_U_R_L` {nok-small}
+|{ok-small} `documentUrl`
+|`DOCUMENT_URL` {ok-small}
+|===
+
+[[unmistakable-abbreviations]]
+=== Unmistakable abbreviations
+
+Avoid abbreviations that are not sufficiently common.
+Seemingly obvious abbreviations in one context can be misunderstood in different contexts.
+
+For example:
+
+|===
+|{nok-small} DON'T |{ok-small} DO
+
+|`temp`
+|Abbreviation unclear; not recommended. (Could mean "temperature" or "temporary".)
+|Abbreviating “minimum” or “maximum” by `min` or `max` is common enough and, therefore, acceptable.
+|`minTemperature` +
+`maxSpeed`
+|===
+
+Using unmistakable abbreviations also applies to <<abbreviations-in-descriptions,Description texts>>.
+
+[[no-redundant-prefix]]
+=== No redundant prefix
+
+Do not duplicate information.
+
+Avoid prefixes that repeat a domain (the namespace) an Aspect Model belongs to.
+
+|===
+|{nok-small} DON'T |{ok-small} DO
+
+|`fooCount` +
+`fooId` +
+`fooXyz` +
+etc. +
+&nbsp; +
+Complete identifier of such a Property: +
+`urn:samm:foo.myapplication:1.0.0#fooCount`
+|`count` +
+`id` +
+`xyz` +
+etc. +
+&nbsp; +
+Complete identifier of such a Property: +
+`urn:samm:foo.myapplication:1.0.0#count`
+|===
+
+NOTE: The complete identifier of a Property is the versioned namespace combined with the element's `name`, for example, `urn:samm:foo.myapplication:1.0.0#count`.
+So, the `foo` context (namespace) is already part of `count`&#8203;'s identifier.
+
+[[no-redundant-suffix]]
+=== No redundant suffix
+
+Do not duplicate information.
+
+Avoid suffixes that repeat the SAMM model element type, such as Aspect, Property, etc.
+
+|===
+|{nok-small} DON'T |{ok-small} DO
+
+|`MovementAspect` +
+&nbsp; +
+Result in TTL file: +
+`:MovementAspect a samm:Aspect ;`
+|`Movement` +
+&nbsp; +
+Result in TTL file: +
+`:Movement a samm:Aspect ;`
+|===
+
+TIP: To distinguish between, for example, an Entity and a related Characteristic, it is acceptable to add corresponding suffixes.
+
+[[meaningful-names]]
+=== Meaningful names (reduce technical debt)
+
+Choose meaningful names for your model elements to foster reuseability and avoid technical debt.
+
+That means:
+
+* Do not simply mirror Property names from data sources like API responses (e.g., JSON).
+** Might be convenient in the short term
+** But makes it difficult to understand and reuse the model in the future
+* Invest the time upfront to choose descriptive names that reflect the purpose and meaning of each element.
+
+TIP: Meaningful names improve code readability and maintainability, whereas arbitrary names make understanding and modifying the model harder in the long run.
+
+Regarding meaningfulness, see also the details given for the `description` attribute at <<no-repeated-preferredname>>.
+
+[[add-payload-names-for-properties]]
+=== Payload names for Properties
+
```

Comment:
If the property name is not suitable as payload name then add a payload name.

In the following some examples for cases you should define a payload name:-
* An expressive Property `name` can be long.

If you consider a Property's name too long for the runtime payload, xref:ROOT:modeling-guidelines.adoc#payload-names[add a payload name].
* conformity to an existing standardized payload the aspect model is derived from (e.g. in IDTA Submodel Templates https://industrialdigitaltwin.org/en/content-hub/submodels property names of ten start with a capital letter)